### ISSUE TITLE
[readme] Should Python 3.10 be mentioned as prerequisite

### ISSUE THREAD

**Issue by kshitij12345** [lcompilers/lpython#2397]
The readme mentions Python 3.10 as a prerequisite but it gets installed whenever the `lp` environment is created with conda (as it is mentioned in the `environment_{unix|windows}.yaml`). 

This may confuse new developers to the project (like me) to have a native Python installation even though it is not required.

I think `conda` should be the only prerequisite.

Thanks!

Ref: https://github.com/lcompilers/lpython#step-0-prerequisites

**Comment by certik** [comment]
@kshitij12345 CPython is not needed for end users of LPython, unless they want to call into CPython.

For developers, we need CPython for the following reasons:

* We have some Python scripts to generate C++ code for AST/ASR
* Our test runners (both tests in `tests` and `integration_tests`) are implemented in Python
* We test every test in `integration_tests` in CPython, to ensure it is compatible with it

So you need CPython for developing LPython.

**Comment by kshitij12345** [comment]
Right, I understand that we need CPython for development.

What I meant was while reading the below instruction
<img width="1117" alt="image" src="https://github.com/lcompilers/lpython/assets/19503980/a5a188e7-8e47-4c06-a303-f0bac6970f14">

I installed Python 3.10 natively on my system but then in the step below, we do install the required Python 3.10 using `conda` for the development environment. So essentially the manual installation I did was not required.
```bash
conda env create -f environment_unix.yml
conda activate lp
```

I guess, this may not be a common issue and maybe I followed the instructions incorrectly 😅 , but the instruction seemed a bit confusing to me.

Thanks!
